extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -438,9 +447,25 @@ public WebDeploymentVerticle(int port, int httpsPort, String host, HttpServerOpt
 
         @Override
         public void start(Future<Void> startFuture) {
+            if (httpOptions == null && httpsOptions == null) {
+                throw new RuntimeException("No HTTPS options specified and HTTP has been disabled");
+            }
+
             final AtomicInteger remainingCount = new AtomicInteger(httpsOptions != null ? 2 : 1);
-            httpServer = vertx.createHttpServer(httpOptions);
-            httpServer.requestHandler(ACTUAL_ROOT);
+            if (httpOptions != null) {
+                httpServer = vertx.createHttpServer(httpOptions);
+                httpServer.requestHandler(ACTUAL_ROOT);
+            } else {
+                httpServer = vertx.createHttpServer(new HttpServerOptions().setPort(port).setHost(host))
+                        .requestHandler(req -> {
+                            req.response()
+                                    .setStatusCode(301)
+                                    .putHeader("Location",
+                                            req.absoluteURI().replace("http", "https").replace(Integer.toString(port),
```

Comment:
This is a more general issue that we need to address but this will fail if you are behind a reverse proxy. This needs to be addressed in a more general way however.